R/flatten_data.R

```diff
@@ -14,16 +15,54 @@
 #' @export
 #'
 #' @examples 
-#' dataset <- ants_L1
+#' flat <- flatten_data(tables = ants_L1[[1]]$tables)
+#' flat
+#' 
+#' flat <- flatten_data(dataset = ants_L1)
+#' flat
+#' 
+flatten_data <- function(
+  tables = NULL, 
+  dataset = NULL){
+  
+  if(!is.null(tables)){
+    return(flatten_tables(tables))
+  }else if(!is.null(dataset)){
+    return(flatten_dataset(dataset))
+  }else{
+    stop("please provide data to flatten")
+  }
+  
+}
+
+
+
+#' @describeIn flatten_data Flatten an ecocomDP formatted dataset
+#' 
+#' @examples 
+#' flat <- flatten_dataset(ants_L1)
+#' flat
 #' 
-#' flat <- flatten_data(dataset[[1]]$tables)
+#' @export
```

Comment:
Only `flatten_data()`, with the `tables` and `dataset` arguments should be exported so users have one consistent interface to the flattening process. It's fine to keep `flatten_dataset()` and `flatten_tables()` in the code base but don't export them.